cmake/TaichiCore.cmake

```diff
@@ -228,6 +229,17 @@ if (TI_WITH_CUDA)
     target_link_libraries(${LIBRARY_NAME} ${llvm_ptx_libs})
 endif()
 
+if (TI_WITH_CUDA_TOOLKIT)
+    message("TI_WITH_CUDA_TOOLKIT ON")
+    message("CUDA_TOOLKIT_ROOT_DIR=$ENV{CUDA_TOOLKIT_ROOT_DIR}")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DTI_WITH_CUDA_TOOLKIT")
```

Comment:
Maybe error out if `TI_WITH_CUDA_TOOLKIT` is on && `CUDA_TOOLKIT_ROOT_DIR` is not set?
Also there might be a better way to do it (although I haven't tested it :P ) https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#module:FindCUDAToolkit

Reply:
I tried before this PR: put `FindCUDA.cmake` in the `cmake/utils` directory and `include(cmake/utils/FindCUDA.cmake)`  in `CMakeLists.txt` like TVM did, but it didn’t work then. ( the name `CUDA_TOOLKIT_ROOT_DIR` comes from cmake :P )

Reply:
It's a great idea and i'll eventually work out a solution to this problem.